Source/WebCore/rendering/style/StyleGradientImage.cpp

```diff
@@ -945,11 +945,15 @@ template<typename GradientAdapter, typename Stops> GradientColorStops StyleGradi
     if (stops.size() > 1 && (*stops.first().offset < 0 || *stops.last().offset > 1))
         gradientAdapter.normalizeStopsAndEndpointsOutsideRange(stops, m_colorInterpolationMethod.method);
     
-    return GradientColorStops::Sorted {
-        stops.template map<GradientColorStops::StopVector>([](auto& stop) -> GradientColorStop {
-            return { *stop.offset, stop.color };
-        })
-    };
+    auto result = stops.template map<GradientColorStops::StopVector>([](auto& stop) -> GradientColorStop {
+        return { *stop.offset, stop.color };
+    });
+
+    std::ranges::stable_sort(result, [](const auto& a, const auto& b) {
```

Comment:
Explicitly sorting seems reasonable, but the assumption has been they were already sorted. Do you know what causes the implicit sort to fail?

Reply:
I've finally figured it out. The actual issue is on the line https://github.com/WebKit/WebKit/blob/525fe6257f8625fbb17f24c28a06ff232490b76d/Source/WebCore/rendering/style/StyleGradientImage.cpp#L807
While the formula is correct, dealing with floats can introduce imprecision problems.
If you add a debug line like this:
```cpp
fprintf(stderr, "newStops[%zu].offset = offset1 + (offset - offset1) * (7 + y) / 13 = %f + %f * %lu = %f + %f = %f\n", y, offset1, (offset - offset1), (7 + y) / 13, offset1, (offset - offset1) * (7 + y) / 13, *newStops[y].offset);
```
you'll see the issue:
```
newStops[6].offset = offset1 + (offset - offset1) * (7 + y) / 13 = -15403597.000000 + 15403597.000000 * 1 = -15403597.000000 + 15403598.000000 = 1.000000
```
Luckily, we don't have to recompute the offset of midpoins. We can simply reuse the original value. I'll update the PR accordingly.